sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/models/CustomHierarchicalListingDeserializer.java

```diff
@@ -17,11 +16,11 @@
 
     @Override
     public BlobHierarchyListSegment deserialize(JsonParser p, DeserializationContext ctxt) throws IOException {
-        ArrayList<BlobItem> blobItems = new ArrayList<>();
+        ArrayList<BlobItemInternal> blobItems = new ArrayList<>();
```

Comment:
Why this rename is required?

Reply:
Because the BlobItem type produced by the generator has a really bad interface for interacting with tags resulting from how it's structured in the payload/swagger. We want to get rid of that and just return a map, which required renaming the generated type to be internal